### PR TITLE
Added some extra logging when OOM occurs in BnP.

### PR DESCRIPTION
**Pull request by FelixGV**
The AvroStoreBuilderMapper can OOM when manipulating certain bad Avro records.

This change does not actually prevent the OOM, but merely prints some useful
info before dying.

### CODE REVIEW

File: contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/AvroStoreBuilderMapper.java [hunk 1]
```diff
@@ -112,11 +115,33 @@ public void map(GenericData.Record record,
                     AvroCollector<Pair<ByteBuffer, ByteBuffer>> collector,
                     Reporter reporter) throws IOException {
 
-        byte[] keyBytes = keySerializer.toBytes(record.get(keyField));
-        byte[] valBytes = valueSerializer.toBytes(record.get(valField));
+        byte[] keyBytes = null, valBytes = null;
+        Object keyRecord = null, valRecord = null;
```
Comment (mattwisein) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/AvroStoreBuilderMapper.java]: The LI style guide recommends doing each declaration on it's own line.

Comment (FelixGV) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/AvroStoreBuilderMapper.java]: Voldemort predates the LI style guide and generally doesn't follow it, but sure. I can split those lines.

**Comment by FelixGV** [comment]
@mattwisein: The removed code is unused cruft.The job was tested...

**Comment by mattwisein** [comment]
Yeah, I have a preference for minimizing the changes we make, especially given the limited testing we have around BnP but I'll leave it to you to decide whether to keep the serializer changes or not.  Other than that it looks good!

**Comment by FelixGV** [comment]
Thanks.

**Pull request closed without merging**